### PR TITLE
Threaded binary tree

### PR DESCRIPTION
**Pull request by amitnayakblr**
Added threaded binary tree, which is a variant of binary trees in the data_structures folder.

### CODE REVIEW

File: data_structures/binary_trees/Threaded_Binary_Tree.c [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+			else if(prev->rlink==temp)
+			{
+				free(prev->rlink);
+				prev->rlink=replacement;
+			}
+		}
+	}
+
+
+void main()
```
Comment (kvedala) [inline data_structures/binary_trees/Threaded_Binary_Tree.c:212]: `main` should return `int` type

File: data_structures/binary_trees/Threaded_Binary_Tree.c [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+	if(curr!=NULL)
+	{
+		printf("%d\t",curr->data);
+		preorder_BT(curr->llink);
+		preorder_BT(curr->rlink);
+	}
+}
+
+// deletes the node if present, else it takes no action.
+void delete_BT(node **root,int ele)
```
Comment (kvedala) [inline data_structures/binary_trees/Threaded_Binary_Tree.c:135]: filenames and function names should all be in *small caps* and without spaces

File: data_structures/binary_trees/threaded_binary_trees.c [hunk 3]
```diff
@@ -0,0 +1,243 @@
+/**
+ * @file
+ * \brief Threaded Binary Tree is a binary tree variant in which all left child
```
Comment (kvedala) [inline data_structures/binary_trees/threaded_binary_trees.c:3]: 👍 on making the fixes

please keep the `\brief` a simple one line and the details separated by a blank like as such:
```c
/**
 * <other tags if any >
 * \brief a one line info of this program
 *
 * a much more elaborate description
 * goes here
 * and 
 * can span many lines
 * <other tags if any>
 */
```

File: data_structures/binary_trees/threaded_binary_trees.c [hunk 4]
```diff
@@ -1,15 +1,13 @@
 /**
  * @file
- * \brief Threaded Binary Tree is a binary tree variant in which all left child
- * pointers that are NULL (in Linked list representation) point to its
- * in-order predecessor, and all right child pointers that are NULL
```
Comment (kvedala) [inline data_structures/binary_trees/threaded_binary_trees.c]: accidentally deleted the details 😨

File: data_structures/binary_trees/threaded_binary_trees.c [hunk 5]
```diff
@@ -1,15 +1,13 @@
...
+ * \brief This file is a simple implementation of a Threaded Binary Tree
+ *
+ * It has the following functionalities:
  * - Insertion
  * - Search
  * - Deletion
  * - Listing of node keys inorder,preorder,postorder
+ *
  * \author [Amitha Nayak](https://github.com/amitnayakblr)
  */
```
Comment (kvedala) [inline data_structures/binary_trees/threaded_binary_trees.c:12]: you can add reference to the other file in the repository:
```c
/**
 * -see binary_search_tree.c
 */
```

File: data_structures/binary_trees/threaded_binary_trees.c (hunk 4 above)
Comment (amitnayakblr) [reply data_structures/binary_trees/threaded_binary_trees.c]: I thought that was unnecessary, added them now.

**Comment by ayaankhan98** [comment]
yeah sure

>  Thank you for your contribution
> @ayaankhan98 you can merge this

yeah sure

**Pull request merged**